sdk/trace/simple_span_processor.go

```diff
@@ -54,8 +54,8 @@ func (ssp *simpleSpanProcessor) OnStart(context.Context, ReadWriteSpan) {}
 
 // OnEnd immediately exports a ReadOnlySpan.
 func (ssp *simpleSpanProcessor) OnEnd(s ReadOnlySpan) {
-	ssp.exporterMu.RLock()
-	defer ssp.exporterMu.RUnlock()
+	ssp.exporterMu.Lock()
+	defer ssp.exporterMu.Unlock()
```

Comment:
This is the only use of `RLock` and `RUnlock`. So if we keep this approach, the `RWMutex` can be changed to a `Mutex`.

I'm not sure this is the proper approach though. The jaeger exporter should be able to run concurrently (maybe it's missing a mutex somewhere?), ensuring it shouldn't be the span processor's responsibility.

Reply:
PTAL https://github.com/open-telemetry/opentelemetry-go/issues/3950